src/numerics/vector_expression.hpp

```diff
@@ -93,6 +93,46 @@ void evaluate(const VectorExpr<T>& expr, mfem::Vector& result)
   }
 }
 
+/**
+ * @brief Fully evaluates a vector expression into an actual mfem::Vector
+ * @param expr The expression to evaluate
+ * @param vec The vector to populate with the expression result
+ * @param comm The MPI_Comm to use for parallel evaluation
+ * @note The performance in a release build is comparable to the serial
+ * implementation, though because this populates the vector on all ranks
+ * the cost of the data copies may exceed any parallelization benefit
+ * for the actual calculation
+ */
+template <typename T>
+void evaluate(const VectorExpr<T>& expr, mfem::Vector& result, MPI_Comm comm)
+{
+  const auto SIZE = expr.Size();
+  SLIC_ERROR_IF(SIZE != static_cast<std::size_t>(result.Size()), "Vector sizes in expression assignment must be equal");
+  int num_procs = 0;
+  int rank      = 0;
+  MPI_Comm_size(comm, &num_procs);
+  MPI_Comm_rank(comm, &rank);
+
+  double* result_arr = result;
+
+  // If the array size is not divisible by # elements, add one
+  const long long per_proc = (SIZE / num_procs) + (SIZE % num_procs != 0) ? 1 : 0;
+
+  // Truncate the number of elements for the last process
+  const long long n_entries = (rank == num_procs - 1) ? SIZE - ((num_procs - 1) * per_proc) : per_proc;
```

Comment:
@joshessman-llnl it seems to me `n_entries` can be negative with this sequence. It sure looks like an edge case, but:
- if SIZE = 1000 and num_procs = 100, then per_proc = 11, and the last process would receive n_entries = -89.

I found it because usually when partitioning the trick to preserve load-balance is to split the residual among the first "residual"-number of procs.

I _think_ those two lines can be substituted by:
```
const long long per_proc = (SIZE / num_procs) + (SIZE % num_procs >= rank + 1) ? 1 : 0;
```

I wanted to investigate it further for fun, but I’m running out of for-fun-time.